presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -154,13 +192,17 @@ private static boolean isJoinEqualityCondition(Expression expression)
                     .filter(subSet -> subSet.size() < numbers.size());
         }
 
-        public static JoinNode createJoinAccordingToPartitioning(JoinGraphNode joinGraph, Set<Integer> partitioning, PlanNodeIdAllocator idAllocator)
+        public Optional<JoinNode> createJoinAccordingToPartitioning(JoinGraphNode joinGraph, Set<Integer> partitioning, PlanNodeIdAllocator idAllocator)
```

Comment:
make package private and `@VisibleForTesting` (and also apply the same schema for other such methods).